collections/_posts/2023-03-13-Fourier_Conv.md

```diff
@@ -0,0 +1,108 @@
+
+&nbsp;
+
+# Fast Fourier Convolution
+
+Let's consider an input of a convolution layer $$X \in \mathbb{R}^{H \times W \times C}$$ and split it along the features' dimension : $$X = \left\{X^{l}, X^{g} \right\}$$, with $$X^{l} \in \mathbb{R}^{H \times W \times (1-\alpha_{in})C}$$ and $$X^{g} \in \mathbb{R}^{H \times W \times \alpha_{in}C}$$. The ratio $$1-\alpha_{in}$$ defines how many of the features will be dedicated to local operations via classic convolutions.
+
+The output of the forward pass of a Fast Fourier Convolution is defined as follows (see Figure 1 for illustration) :
+
+$$
+Y^{l} = Y^{l \rightarrow l} + Y^{g \rightarrow l} = f_{l}(X^{l}) + f_{g \rightarrow l}(X^{g}) \\
+Y^{g} = Y^{g \rightarrow g} + Y^{l \rightarrow g} = f_{g}(X^{g}) + f_{l \rightarrow g}(X^{l})
+$$
+
+
+where $$Y^{l} \in \mathbb{R}^{H \times W \times (1-\alpha_{out})C}$$ and $$Y^{l} \in \mathbb{R}^{H \times W \times \alpha_{out}C}$$.  To simplify the hyperparameters' tuning, the two ratios $$\alpha_{in}$$ and $$\alpha_{out}$$ are set equal.
+
+$$f_{l}(\cdot)$$ is aimed to capture small scale information and $$f_{g \rightarrow l}(\cdot)$$ and $$f_{}(\cdot)$$ serve to exchange information between local and global levels. These three components can be any regular convolutional blocks, for instance 3x3 Conv + BN + ReLU. The term $$f_{g}(\cdot)$$, called *spectral transformer*, is operating at a global scale though Fourier transform, and needs to be further explained. 
+
+<div style="text-align:center">
+<img src="/collections/images/fourier_conv/ffc.jpg" height=400></div>
+<p style="text-align: center;font-style:italic">Figure 1. Architecture of a Fast Fourier Convolution.</p>
+
+
+&nbsp;
+
+## Fourier Unit and Local Fourier Unit
+
+**Fourier Unit**
+
+The goal of the global path in the FFC is to enlarge the receptive field of the convolution to the full resolution of the input feature map. To do so, the input channels are passed to the spectral domain with a discrete Fourier transform. When aplying 2-D Fourier Transform on some real signals, it returns a Hermitian matrix, and convertly, applying inverse 2-D Fourier Transform on an Hermitian Matrix results in a matrix with only real elements. These two principles are used to ensure compatibily with other real neural layers, as well as allowing to retain only half on the information of the spectral signal.
+
+In the spectral domain, the real and imaginary parts of the signal are concatenated and standard convolutions layers can be used on the new feature tensor. Since a change in any region of the image in the original space has an influence on the whole spectral domain, unsing 1x1 convolution after a Fourier transform is sufficient to have a global receptive field. 
```

Comment:
```suggestion
In the spectral domain, the real and imaginary parts of the signal are concatenated and standard convolutions layers can be used on the new feature tensor. Since a change in any region of the image in the original space has an influence on the whole spectral domain, using 1x1 convolution after a Fourier transform is sufficient to have a global receptive field. 
```